### PR TITLE
add numpy function support in Groupby.agg

### PR DESCRIPTION
**Pull request by RainFung**
Everything is OK except the return type infer. @HyukjinKwon Can help me fix it ?

### CODE REVIEW

File: databricks/koalas/groupby.py [hunk 1]
```diff
@@ -128,14 +137,36 @@ def aggregate(self, func_or_funcs, *args, **kwargs):
...
+                if isinstance(aggfunc, Callable):
+                    func_name = aggfunc.__name__
+                    data_col = "('{0}', '{1}')".format(key, func_name) if multi_aggs else key
+                    data_columns.append(data_col)
+                    column_index.append((key, func_name))
+
+                    # Here we execute with the first 1000 to get the return type.
+                    # If the records were less than 1000,
+                    # it uses pandas API directly for a shortcut.
+                    limit = 1000
```
Comment (HyukjinKwon) [inline databricks/koalas/groupby.py]: You can do, for instance, as below:

```python
assert callable(func), "the first argument should be a callable function."

spec = inspect.getfullargspec(func)
return_sig = spec.annotations.get("return", None)

# If return type hint is not found, we should try to infer schema after executing
# pandas native function.
should_infer_schema = return_sig is None

if should_infer_schema:
    # TODO: this limit should be configurable.
    limit = 1000

    # Collect top n and conpute to infer the return type.
    pdf = self._kdf.head(limit + 1)._to_internal_pandas()
    grouped_pdf = pdf.groupby(groupkeys_name).agg({key: aggfunc})

    if len(pdf) <= limit:
        return Series(grouped_pdf)

    # If collected pandas DataFrame is larger than limit, we just use its type.
    return_type =  _infer_return_type(grouped_pdf.dtypes[0].type)

    pandas_func = F.pandas_udf(aggfunc, returnType=return_type,
                               functionType=F.PandasUDFType.GROUPED_AGG)
else:
    # In this case, we just use the type specified in type hint at the given function.
    return_type = _infer_return_type(func).tpe

pandas_udf(aggfunc, return_type, functionType=F.PandasUDFType.GROUPED_AGG)
```

I haven't tested but it should work.

**Comment by HyukjinKwon** [comment]
gentle ping @RainFung, are you still working on this?

**Comment by RainFung** [comment]
Sorry, i'm busy recently. I'll work it in this weekend.

**Comment by HyukjinKwon** [comment]
Thanks @RainFung!

**Comment by ueshin** [comment]
@RainFung Do you want to continue working on this?
If you don't have time, shall we close this for now and someone else might want to work on this.
Thanks!

**Comment by RainFung** [comment]
I can back to it.

**Comment by RainFung** [comment]
Im back to it.

**Comment by RainFung** [comment]
@HyukjinKwon Sorry, Sir. There are too many problems when i solve this PR. I'm afraid I can't finish it.Can you complete this PR?

**Comment by HyukjinKwon** [comment]
Yeah, that's fine. I will take this over when I find some time. Let's leave this closed for now.

**Pull request closed without merging**